ds-prep-python-data-series/list_tuples.py

```diff
@@ -0,0 +1,162 @@
+
+def last_appearing_rate(keys, values):
+  last_data_point = values[-1]
+  dictionary = {
+    keys[0] : last_data_point[0],
+    keys[1] : last_data_point[1]
+  }
+  return dictionary
+
+print(last_appearing_rate(columns, unemployment_rates))
+print("---")
+
+# Get the first five series points, in the order they appear, as a list of
+# tuples.
+def first_five_series_points(series):
+  return series[0:4]
+
+print(first_five_series_points(unemployment_rates))
+print("---")
+
+# Check if the years 2000 and 2010 are each included in the data series.
+def check_years_present(value, data_series):
+  list_of_years = []
+  for x in data_series:
+    (year_value, rate_value) = x
+    list_of_years.append(year_value)
+  return value in list_of_years
+
+print(check_years_present(2000, unemployment_rates))
+print(check_years_present(2010, unemployment_rates))
+print("---")
+
+# Get the unemployment rate for the most recent year.
+
+def ue_rate(data_series):
+  data_series.sort()
+  year_data = data_series[-1]
+  return year_data[1]
+
+print(ue_rate(unemployment_rates))
+print("---")
+
+# Get all the unique years included in the series, as a set.
+
+def unique_years(data_series):
+  list_of_years = []
+  for x in data_series:
+    (year_value, rate_value) = x
+    list_of_years.append(year_value)
+  unique_years = set(list_of_years)
+  return unique_years
+
+print(unique_years(unemployment_rates))
+print("---")
+
+# Get the list of the unemployment rates, ordered by year.
+def ordered_by_year(data_series):
+  data_series.sort()
+  return data_series
+
+print(ordered_by_year(unemployment_rates))
+print("---")
+
+# Get the largest unemployment rate.
+def largest_ue_rate(data_series):
+  list_of_rates = []
+  for x in data_series:
+    (year_value, rate_value) = x
+    list_of_rates.append(rate_value)
+  largest_ue_rate = max(list_of_rates)
+  return largest_ue_rate
+
+print(largest_ue_rate(unemployment_rates))
+print("---")
+
+# Create a new series, in the same format as the original, with the
+# "employment rate" instead of the "unemployment rate" (the rates are listed in
+# percentages).
+
+def employment_rate(data_series):
+  years_list = [] # define list of years
+  ue_rates_list = [] # define list of ue rates
+  e_rates_list = [] # define list of e rates
+  merged_list = [] # define new merged list
+  for x in data_series: # run a loop on the data series
+    (year_value, rate_value) = x # unpack the tuple
+    years_list.append(year_value) # append year values to list
+    ue_rates_list.append(rate_value) # append ue values to list
+  for y in ue_rates_list: # run a loop on the ue values
+    e_rate = 100 - y # calculate the e value
+    e_rates_list.append(e_rate) # append the e values to the e list
+  merged_list = [(years_list[i], e_rates_list[i]) for i in range(0, len(years_list))] # merge the lists
+  return merged_list # return merged list
+
+print(employment_rate(unemployment_rates))
+print("---")
```

Comment:
Looks like this produces the correct output, though there's a much simpler way to accomplish the same task.